src/runtime/RegExpObject.cpp

```diff
@@ -506,4 +509,16 @@ void RegExpObject::pushBackToRegExpMatchedArray(ExecutionState& state, ArrayObje
         }
     }
 }
+
+bool RegExpObject::checkMatchSymbol(ExecutionState& state)
+{
+    ObjectGetResult symbol = get(state, ObjectPropertyName(state, state.context()->vmInstance()->globalSymbols().match));
+
+    Value val = symbol.value(state, Value());
+
+    if (!val.isUndefined() && val.isBoolean()) {
+        return val.asBoolean();
+    }
```

Comment:
This is incorrect. The standard says: `If isRegExp is not undefined, return ToBoolean(isRegExp).`
so the correct way is:

```c
if (!val.isUndefined()) {
  return val.toBoolean(state);
}
```

Reply:
The ToBoolean operation returns true, for object(function) argument, so I think

```
if (!val.isUndefined()) {
  return val.toBoolean(state);
}
```
is still enough there.